ExpenseSplit/HomeScreen.swift

```diff
@@ -0,0 +1,162 @@
+                        .navigationBarItems(
+                            leading: Button(action: {
+                                showSettleBill = true
+                            }) {
+                                Image(systemName: "line.horizontal.3")
+                                    .imageScale(.large)
+                            },
+                            
+                            trailing: Button(action: {
+                                // Handle cart action
+                            }) {
+                                Image(systemName: "heart.fill")
+                                    .imageScale(.large)
+                                    .overlay(
+                                        Text("2")
+                                            .font(.caption)
+                                            .foregroundColor(.white)
+                                            .padding(5)
+                                            .background(Color.red)
+                                            .clipShape(Circle())
+                                            .offset(x: 10, y: -10)
+                                    )
+                            })
+        }
+    }
+}
+
+struct BalanceSummaryView: View {
+    
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text("Total balance")
+                .font(.headline)
+            HStack {
+                VStack(alignment: .leading) {
+                    Text("You owe")
+                        .foregroundColor(.red)
+                    Text("€92.21")
+                        .foregroundColor(.red)
+                }
+                Spacer()
+                VStack(alignment: .leading) {
+                    Text("You are owed")
+                        .foregroundColor(.green)
+                    Text("$69.77 + ₹1,550.50 + £20.00")
+                        .foregroundColor(.green)
+                }
+            }
+        }
+        .padding()
+        .background(Color.gray.opacity(0.1))
+        .cornerRadius(10)
+        .padding([.leading, .trailing])
+    }
+}
+
+struct FriendRow: View {
+    let friend: Friend
+    
+    var body: some View {
+        HStack {
+            Image(friend.avatar)
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 50, height: 50)
+                .clipShape(Circle())
+                .overlay(Circle().stroke(Color.gray, lineWidth: 1))
+            
+            VStack(alignment: .leading) {
+                Text(friend.name)
+                    .font(.headline)
+                Text(friend.balance)
+                    .foregroundColor(friend.isOwed ? .green : .red)
+            }
+            Spacer()
+            if friend.balance != "settled up" {
+                Text(friend.isOwed ? "owes you" : "you owe")
+                    .foregroundColor(friend.isOwed ? .green : .red)
+            }
+        }
```

Comment:
This view is looking good! With the adjustments I suggested for the Friend model—like making some properties variables and adding computed properties—you could streamline this view even further. By leveraging computed properties, you can make the logic for determining text and colors more dynamic, reducing the need for repetitive checks. This would simplify the code here, making it more concise and easier to maintain. It’s just a suggestion to consider, and I think it could enhance the flexibility of this view. Great job overall!